### PR TITLE
Update Header and Footer

### PR DESCRIPTION
**Pull request by TheShubham99**
#34 

- [x] Add link to Adventure Site.
- [x] Add Patreon link to footer.
- [x] Remove freenode link.

### CODE REVIEW

File: src/components/Footer/Footer.jsx [hunk 1]
```diff
@@ -109,12 +109,9 @@ export default () => (
               <FaRedditSquare />
             </a>
           </IconContext.Provider>
-          <IconContext.Provider value={{ size: "2.5em" }}>
-            <a
-              className="link"
-              href="http://webchat.freenode.net/?channels=terasology&uio=d4?channels=%23terasology&nick=Terasologist...&prompt=1&useUserListIcons=true"
-            >
-              <FaComments />
+          <IconContext.Provider value={{ size: "2em" }}>
```
Comment (majordwarf) [inline src/components/Footer/Footer.jsx:112]: It's nice to see the Patreon link finally being linked but as far as I can see Font Awesome has some inconsistency maintaining similar styling which results in the Patreon logo rendering bigger than other social media icons. I'd recommend using 2.1em instead of 2em as it seems visually more close to the size of other logos.

File: src/components/Footer/Footer.jsx [hunk 2]
```diff
@@ -109,12 +109,9 @@ export default () => (
...
+            <a className="link icon" href="https://www.patreon.com/Terasology">
```
Comment (majordwarf) [inline src/components/Footer/Footer.jsx:113]: You included a new class here to increase padding the reason being discussed above. But a comment above the snippet here would help to learn why the changes took place and why are we not keeping up with the 2.5em size for the logo. This works for now but having a comment explaining would be much helpful for future maintainers.

File: src/layout/css/_icons.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+.icon {
+  padding: 0.2em;
+}
```
Comment (majordwarf) [inline src/layout/css/_icons.scss:3]: A new file just for a class that does changes to footer icons doesn't make much sense to me. I'd prefer the following bit to stay in `footer.scss`.

File: src/layout/css/style.scss [hunk 4]
```diff
@@ -1,9 +1,10 @@
-@import "variables";
-@import "layout";
-@import "typography";
-@import "cards";
-@import "buttons";
-@import "footer";
-@import "header";
-@import "search";
-@import "posts";
```
Comment (majordwarf) [inline src/layout/css/style.scss:9]: Changes like adding a new line should be in its own commit and should not be mixed up.

File: .gitignore [hunk 5]
```diff
@@ -58,6 +58,9 @@ public/
 # Bundle stats
 bundle-stats.json
 
+# Package Lock
+package-lock.json
+
```
Comment (majordwarf) [inline .gitignore:63]: While including this is a good choice, I can see you also deleted package-lock.json in the same commit which was generated by you. A clean commit just including the above would be great to keep the git history clean.

File: src/components/Header/Header.jsx [hunk 6]
```diff
@@ -34,6 +34,9 @@ export default () => {
             <NavItem>
               <NavLink href="/media">Media</NavLink>
             </NavItem>
+            <NavItem>
+              <NavLink href="/AdventureSite">Explore</NavLink>
```
Comment (majordwarf) [inline src/components/Header/Header.jsx:38]: The following link returns `404` and I guess this link should not be included till the moment the Adventure site is live.

**Review by majordwarf — CHANGES_REQUESTED** [review]
@jdrueckert Hey! I just requested some changes, would like to hear your inputs on this. Sorry for the long review but I got a habit of writing fleshed-out commits and PRs this summer @TheShubham99 . 

Also, we should start using some commit conventions as it'll be easy to automate things in the future.

File: src/components/Header/Header.jsx (hunk 6 above)
Comment (jdrueckert) [reply src/components/Header/Header.jsx:38]: The AdventureSite is currently hosted at https://terasology.org/AdventureSite/ so I'd recommend using that link for the time being.

File: src/layout/css/_icons.scss (hunk 3 above)
Comment (TheShubham99) [reply src/layout/css/_icons.scss:3]: I thought of adding this in the `footer.scss` but ended up making a new file. Will fix this 👍

File: src/components/Footer/Footer.jsx (hunk 2 above)
Comment (TheShubham99) [reply src/components/Footer/Footer.jsx:113]: Okay. @majordwarf So, even if we change size to 2.1 em we still need that padding. So if I am not wrong you're expecting a comment explaining the reason of size inconsistency.. right? 🤔

**Comment by jdrueckert** [comment]
hey @TheShubham99 any update on this yet? 🙃
